tests/ValueTest.php

```diff
@@ -224,6 +225,34 @@ public function testFailAsNonEmptyStrings($raw): void
         $value->asNonEmptyStrings();
     }
 
+    /**
+     * @dataProvider dateTimeProvider
+     *
+     * @psalm-param mixed $raw
+     * @psalm-param class-string|null $expectedException
+     */
+    public function testAsDateTimeImmutable($raw, ?string $expectedException, $expected): void
+    {
+        $value = new Value($raw);
+
+        if ($expectedException !== null) {
+            $this->expectException($expectedException);
+        }
+
+        $date = $value->asDateTimeImmutable();
+
+        self::assertEquals($expected, $date->format(\DateTimeInterface::ATOM));
+    }
+
+    public static function dateTimeProvider(): iterable
+    {
+        yield [null, InvalidArgumentException::class, null];
+        yield [1234, InvalidArgumentException::class, null];
+        yield ['blub', TypeError::class, null];
+        yield ['20221111', null, '2022-11-11T00:00:00+00:00'];
+        yield ['2022-11-11', null, '2022-11-11T00:00:00+00:00'];
```

Comment:
Can we add tests which include times and timezones?

Reply:
Added